src/provider/model/AppDirectory.ts

```diff
@@ -231,3 +263,50 @@ export class AppDirectory extends AsyncInit {
 function intentSupportsContext(intent: AppDirIntent, contextType: string): boolean {
     return intent.contexts === undefined || intent.contexts.length === 0 || intent.contexts.includes(contextType);
 }
+
+function filterUrlsByScope(scope: ShardScope, urls: string[]): string[] {
+    return urls.filter((url) => {
+        if (isUrlValidForScope(scope, url)) {
+            return true;
+        } else {
+            if (scope.type === 'domain') {
+                console.warn(`Not including remote snippet at '${url}' in App Directory. URL not in domain '${scope.domain}'`);
+            } else {
+                console.warn(`Not including remote snippet at '${url}' in App Directory`);
+            }
+
+            return false;
+        }
+    });
+}
+
+function filterAppsByScope(scope: ShardScope, applications: Application[]): Application[] {
+    return applications.filter((app) => {
+        if (isUrlValidForScope(scope, app.manifest)) {
+            return true;
+        } else {
+            if (scope.type === 'domain') {
+                console.warn(`Not including application '${app.name}' in App Directory. Manifest URL '${app.manifest}' not in domain '${scope.domain}'`);
+            } else {
+                console.warn(`Not including application '${app.name}' in App Directory`);
+            }
+
+            return false;
+        }
+    });
+}
+
+function isUrlValidForScope(scope: ShardScope, url: string): boolean {
+    if (scope.type === 'global') {
+        return true;
+    } else if (scope.type === 'domain') {
```

Comment:
In new version they're already combined when they come out of the AppDirectoryStore, and even if that wasn't the case I'd end up repeating myself, performing the same step twice, first without the filter then without

Reply:
Should validation be inside `AppDirectoryStore` then?.. To remove invalid data as early as possible, rather than (potentially, if other parts of the service needed to query the directory store) needing to duplicate the validation elsewhere?

Reply:
The point of AppDirectoryStore was to be a light wrapper around the use of `fin`, so we could avoid referencing fin in tests. Moving the config store stuff into AppDirectoryStore I can juuuust about be ok with, but this feels like to much. I also would argue that anywhere using AppDirectoryStore directly is doing something wrong.

This could instead be 3 layers, the fin wrapper layer, the filter and combine with config store layer, and the actual AppDirectory layer, but that feels like overkill

Reply:
Agree three layers is a bit much, makes sense keeping `ADS` as more of a fin-abstraction.